secrets/secrets.go

```diff
@@ -0,0 +1,27 @@
+package secrets
+
+import "sync"
+
+type SecretManager interface {
```

Comment:
the "manager" term is over used and is getting very confusing
can we call this interface as Secrets? may be even rename the methods to Login, Set, Get, CheckLogin, SetDefaultKey? what do you think?

Reply:
If we want to go by the golang idiomatic way, then
this package is already called secrets. So this interface can be called just Manager and the APIs would be Login, Set, Get etc

So the caller of this package would do this
```
secretsManager := secrets.NewManager()
secretsManager.Login
```